features/admin.actions.v1/components/action-config-form.tsx

```diff
@@ -0,0 +1,829 @@
+                            "types.apiKey.properties.value.validations.empty");
+                    }
+                }
+
+                break;
+            default:
+                break;
+        }
+
+        return error;
+    };
+
+    const handleSubmit = async (values: ActionConfigFormPropertyInterface) => {
+        const authProperties: Partial<AuthenticationPropertiesInterface> = {};
+
+        if (isAuthenticationUpdateFormState || isCreateFormState) {
+            switch (authenticationType) {
+                case AuthenticationType.BASIC:
+                    authProperties.username = values.usernameAuthProperty;
+                    authProperties.password = values.passwordAuthProperty;
+
+                    break;
+                case AuthenticationType.BEARER:
+                    authProperties.accessToken = values.accessTokenAuthProperty;
+
+                    break;
+                case AuthenticationType.API_KEY:
+                    authProperties.header = values.headerAuthProperty;
+                    authProperties.value = values.valueAuthProperty;
+
+                    break;
+                case AuthenticationType.NONE:
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (isCreateFormState) {
+            const actionValues: ActionInterface = {
+                endpoint: {
+                    authentication: {
+                        properties: authProperties,
+                        type: authenticationType
+                    },
+                    uri: values.endpointUri
+                },
+                name: values.name
+            };
+
+            setIsSubmitting(true);
+            createAction(actionTypeApiPath, actionValues)
+                .then(() => {
+                    handleSuccess(ActionsConstants.CREATE);
+                    mutateActions();
+                })
+                .catch((error: AxiosError) => {
+                    handleError(error, ActionsConstants.CREATE);
+                })
+                .finally(() => {
+                    setIsSubmitting(false);
+                });
+        } else {
+            // Updating the action
+            const updatingValues: ActionUpdateInterface = {
+                endpoint: isAuthenticationUpdateFormState || values.endpointUri !== initialValues?.endpointUri ? {
+                    authentication: isAuthenticationUpdateFormState ? {
+                        properties: authProperties,
+                        type: authenticationType
+                    } : undefined,
+                    uri: values.endpointUri !== initialValues?.endpointUri ? values.endpointUri : undefined
+                } : undefined,
+                name: values.name !== initialValues?.name ? values.name : undefined
+            };
+
+            setIsSubmitting(true);
+            updateAction(actionTypeApiPath, initialValues?.id, updatingValues)
```

Comment:
Unnecessary null check. If the initialValues are undefined, undefined will be passed down to the updateAction.

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/6722/commits/5e3564e0797c09f4f20b71837b928bc9fa57c847